File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/builder/ConcreteBuilder2.java
package jv.com.tencent.cain.design_pattern.builder;

/**
 * @author : jiangyu
 * @date : 2020/2/10
 * @desc : xxx
 */
public class ConcreteBuilder2 extends Builder {
    private Product product = new Product();

    @Override
    public void buildPartA() {
        product.addPart("部件X");
    }

    @Override
    public void buildPartB() {
        product.addPart("部件Y");

    }

    @Override
    public Product getResult() {
        return product;
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/builder/ConcreteBuilder2.kt
package kt.com.tencent.cain.desigin_pattern.builder

/**
 * @author : jiangyu
 * @date   : 2020/2/10
 * @desc   : xxx
 */
class ConcreteBuilder2(val product: Product = Product()) : Builder() {
    override fun buildPartA() {
        product.parts.add("部件X")
    }

    override fun buildPartB() {
        product.parts.add("部件Y")
    }

    override fun getResult(): Product = product
}